core/client/fs/src/main/java/alluxio/client/file/ufs/UfsBaseFileSystem.java

```diff
@@ -399,6 +399,7 @@ private URIStatus transformStatus(UfsStatus ufsStatus) {
         CommonUtils.stripPrefixIfPresent(ufsStatus.getName(), mRootUFS.getPath())));
     FileInfo info = new FileInfo().setName(ufsUri.getName())
         .setPath(ufsStatus.getName())
+        .setFileId(ufsUri.hashCode())
```

Comment:
Some of the fields may be optional, but the FileId should be a must.
We should check and assert this in build().

Reply:
Yeah FileInfo is better to be a builder with arguments check to force set needed fields.
But currently FileInfo allows field modification for serialization and deserialization.

Need to implement a customized serializer that supports the new fileInfo without setters.
Prestodb has examples of how to implement it.
I previously consider change it but it's a too big of change that i didn't get time to. And this is one item on the 3.0 best practice list.